hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/QueryHandler.java

```diff
@@ -65,6 +66,19 @@ public interface QueryHandler {
      */
     boolean needsAnswerOnlyCost(@NonNull final ResponseType responseType);
 
+    /**
+     * Computes the fees associated with this given query. This method is called during the query workflow, before
+     * validation, and used to determine whether the associated payment is sufficient, and whether the payer has
+     * sufficient funds.
+     *
+     * @param queryContext The context for the query being handled
+     * @return The fees associated with the query
+     */
+    @NonNull
+    default Fees computeFees(@NonNull QueryContext queryContext) {
```

Comment:
Should this not be default and is implemented as FREE in `FreeQueryHandler`?

Reply:
Yes, that is probably the right way to do it. I have another issue to follow upon this and implement query fee computation properly.